src/dune_rules/melange_rules.ml

```diff
@@ -16,18 +16,31 @@ let ocaml_flags sctx ~dir melange =
     with_vendored_flags ~ocaml_version flags
   | false -> flags
 
-let lib_output_dir ~target_dir lib =
+let lib_output_dir ~sctx ~target_dir lib =
   let info = Lib.info lib in
+  let local () =
+    let lib = Lib.Local.of_lib_exn lib in
+    let info = Lib.Local.info lib in
+    Lib_info.src_dir info
+  in
   let lib_dir =
-    match Lib.Local.of_lib lib with
-    | None ->
+    match Lib_info.status info with
+    | Private _ -> local ()
```

Comment:
can local be inlined here?

Reply:
Done in https://github.com/anmonteiro/dune/pull/4/commits/0ebe416203a6bf35fb4cdd4fac867a97802ddc1d.